csharp/src/Apache.Arrow.Flight.Sql/Client/FlightSqlClient.cs

```diff
@@ -0,0 +1,1191 @@
+                var commandSqlCall = new CommandPreparedStatementQuery
+                {
+                    PreparedStatementHandle = preparedStatementResponse.PreparedStatementHandle
+                };
+
+                byte[] commandSqlCallPackedAndSerialized = commandSqlCall.PackAndSerialize();
+                var descriptor = FlightDescriptor.CreateCommandDescriptor(commandSqlCallPackedAndSerialized);
+                return await GetFlightInfoAsync(options, descriptor).ConfigureAwait(false);
+            }
+
+            throw new InvalidOperationException("No results returned from the query.");
+        }
+        catch (RpcException ex)
+        {
+            throw new InvalidOperationException("Failed to execute query", ex);
+        }
+    }
+
+    /// <summary>
+    /// Executes an update query on the server.
+    /// </summary>
+    /// <param name="options">RPC-layer hints for this call.</param>
+    /// <param name="query">The UTF8-encoded SQL query to be executed.</param>
+    /// <param name="transaction">A transaction to associate this query with. Defaults to no transaction if not provided.</param>
+    /// <returns>The number of rows affected by the operation.</returns>
+    public async Task<long> ExecuteUpdateAsync(FlightCallOptions options, string query, Transaction? transaction = null)
+    {
+        transaction ??= Transaction.NoTransaction;
+
+        if (options == null)
+        {
+            throw new ArgumentNullException(nameof(options));
+        }
+
+        if (string.IsNullOrEmpty(query))
+        {
+            throw new ArgumentException("Query cannot be null or empty", nameof(query));
+        }
+
+        try
+        {
+            var updateRequestCommand =
+                new ActionCreatePreparedStatementRequest { Query = query, TransactionId = transaction.TransactionId };
+            byte[] serializedUpdateRequestCommand = updateRequestCommand.PackAndSerialize();
+            var action = new FlightAction(SqlAction.CreateRequest, serializedUpdateRequestCommand);
+            var call = DoActionAsync(options, action);
+            long affectedRows = 0;
+
+            await foreach (var result in call.ConfigureAwait(false))
+            {
+                var preparedStatementResponse = result.Body.ParseAndUnpack<ActionCreatePreparedStatementResult>();
+                var command = new CommandPreparedStatementQuery
+                {
+                    PreparedStatementHandle = preparedStatementResponse.PreparedStatementHandle
+                };
+
+                var descriptor = FlightDescriptor.CreateCommandDescriptor(command.PackAndSerialize());
+                var flightInfo = await GetFlightInfoAsync(options, descriptor);
+                var doGetResult = DoGetAsync(options, flightInfo.Endpoints[0].Ticket);
+                await foreach (var recordBatch in doGetResult.ConfigureAwait(false))
```

Comment:
This is missing a `using` or a call to `Dispose`.  Please ensure all `IDisposable` items are properly disposed in the client and prepared statement code paths when they are not returned to the caller.